bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Device.java

```diff
@@ -820,26 +826,30 @@ void new_Object (Object object) {
 		for (int i=0; i<objects.length; i++) {
 			if (objects [i] == null) {
 				objects [i] = object;
-				errors [i] = new Error ();
+				if (tracking) {
+					errors [i] = new Error ();
+				}
 				return;
 			}
 		}
 		Object [] newObjects = new Object [objects.length + 128];
 		System.arraycopy (objects, 0, newObjects, 0, objects.length);
 		newObjects [objects.length] = object;
 		objects = newObjects;
-		Error [] newErrors = new Error [errors.length + 128];
-		System.arraycopy (errors, 0, newErrors, 0, errors.length);
-		newErrors [errors.length] = new Error ();
-		errors = newErrors;
+		if (tracking) {
+			Error [] newErrors = new Error [errors.length + 128];
+			System.arraycopy (errors, 0, newErrors, 0, errors.length);
+			newErrors [errors.length] = new Error ();
+			errors = newErrors;
+		}
 	}
 }
 
 void printErrors () {
 	if (!DEBUG) return;
 	if (tracking) {
 		synchronized (trackingLock) {
-			if (objects == null || errors == null) return;
+			if (objects == null && errors == null) return;
```

Comment:
Why the change here?

Reply:
Its because earlier we had both objects and errors guarded by the lock. But now, object is not guarded by the lock anymore and it would not be null anymore. So, we only wanna return already if both image and errors are null. i think even one check just for errors would also be fine. We don't need to check for objects since the method is called printErrors. Why print only objects in case there's no error? What do you say?

Reply:
The change has been reverted.